packager/media/formats/mp4/multi_segment_segmenter.cc

```diff
@@ -150,6 +150,7 @@ Status MultiSegmentSegmenter::WriteSegment() {
   } else {
     file_name = GetSegmentName(options().segment_template,
                                sidx()->earliest_presentation_time,
+                               moof()->tracks[0].decode_time.decode_time,
```

Comment:
This is the crux of it. I'm pretty sure hardcoding the `[0]` index is a logical bug, but am still grokking the codebase. Also I want to see if I can avoid needing a `moof()` method at all.